R/AddCardModule.R

```diff
@@ -110,19 +110,35 @@ add_card_button_srv <- function(id, reporter, card_fun) {
           ReportCard$new(),
           eval(default_card, envir = environment(card_fun))
         )
-        if (length(card_fun_args_nams) == 1) {
-          card <- card_fun(card)
-          if (length(input$comment) > 0 && input$comment != "") {
-            card$append_text("Comment", "header3")
-            card$append_text(input$comment)
+        has_own_comment <- length(card_fun_args_nams) == 2
+        if (has_own_comment) {
+          card <- try(card_fun(card, input$comment))
+        } else {
+          card <- try(card_fun(card))
+          if (!inherits(card, "try-error")) {
+            if (length(input$comment) > 0 && input$comment != "") {
+              card$append_text("Comment", "header3")
+              card$append_text(input$comment)
+            }
           }
+        }
+
+        if (inherits(card, "try-error")) {
+          msg <- paste0(
+            "The card was not added successfully.",
+            "Possibly the card_fun is invalid or the shiny process is not ready to be catched."
```

Comment:
```suggestion
            "The card could not be added to the report. ",
            "Have the outputs for the report been created yet? If not please try again when they ",
            "are ready. Otherwise contact your application developer" 
```